### PR TITLE
Create job from job definition - UI

### LINKED ISSUE #209: Run now from a job definition
### Problem

Previously, we offered a way for users to "re-run" a previously run job, but this didn't leverage a job definition. We removed that feature as it was incompatible with the API change we were making (removing the `input_url`). However, we believe that users we will still want to rerun a job. 

### Proposed Solution

We proposed allowing users to "re-run" a job by adding a "Run now" action for a job definition. To implement this we would create a new REST API and `Scheduler` method and then build the UI on top of this. The reason we are thinking of using the job definition for this is as follows:

* The job definition already has an immutable snapshot of the input file that can be used for the job.
* The job definition allows us to show the user all of the jobs that are "grouped" by the job definition.

### LINKED ISSUE #225: "Create Job" form title is not localized
The title of the "Create Job" form is `Create Job`, not `trans.__('Create Job')`.

https://github.com/jupyter-server/jupyter-scheduler/blob/03cb4e9b8f05f57618860d5621398157aa92092d/src/mainviews/create-job.tsx#L756

### PR DESCRIPTION
**Pull request by jweill-aws**
Fixes #209 and #225.

On the Job Definition Detail page, adds a "Run Job" button that brings the user to a minimal version of the Create Job form that displays only the job definition ID and lets the user edit parameters. The API to create a job this way was added in #228.

Demo video:

https://user-images.githubusercontent.com/93281816/198753293-50c90be5-7e93-4307-879a-0ccb52816195.mov

### CODE REVIEW

**Comment by ellisonbg** [comment]
Overall, looks good. Wondering if we might want to name the action "Run job now" to emphasize that this is different from running it on a schedule. Also, the API in #228 will only allow users to set parameters, so we should hide the Additional options in this form. Also, will this form have the same job creation error handling as the normal create job (show the spinner, hide the buttons, show the error banner)?

**Comment by jweill-aws** [comment]
I hesitated to use the word “now” because the button doesn’t take an immediate action, but rather brings the user to a form. (Perhaps “Run Job…” or “Run Job Now…” would be better.)

The button hiding behavior and error handling is the same as in the ordinary create job/JD form.

Sent from my iPhone

On Oct 29, 2022, at 12:12 PM, Brian E. Granger ***@***.***> wrote:

﻿

Overall, looks good. Wondering if we might want to name the action "Run job now" to emphasize that this is different from running it on a schedule. Also, the API in #228<https://github.com/jupyter-server/jupyter-scheduler/pull/228> will only allow users to set parameters, so we should hide the Additional options in this form. Also, will this form have the same job creation error handling as the normal create job (show the spinner, hide the buttons, show the error banner)?

**Comment by jweill-aws** [comment]
(Apologies for the botched non-rebase; I hadn't set up Git properly on my new computer yet)

Updates this change to remove advanced options from the run-from-JD form. Still WIP because no API is available yet.

image

**Comment by jweill-aws** [comment]
This change is blocked until #228 is merged.

**Comment by 3coins** [comment]
Request payload seems incorrect here, so passed parameters are not assigned. The payload should match the `CreateJobFromDefinition` model.
https://github.com/jupyter-server/jupyter-scheduler/blob/9dd885c967c98ee9bceb8046c066d1b2e5411de5/jupyter_scheduler/models.py#L272-L273

**Review by 3coins — CHANGES_REQUESTED** [review]

**Review by 3coins — APPROVED** [review]

**Pull request merged**
